test/src/unit-max-tile-size.cc

```diff
@@ -0,0 +1,1116 @@
+    CHECK(c_get_fragments_max_in_memory_tile_size(main_array_name_) == 257);
+
+    array_schema_evolve_B();
+    CHECK(c_get_fragments_max_in_memory_tile_size(main_array_name_) == 257);
+    write_dense_array_attr_nchar_ntimes<42>(main_array_name_, "c42");
+    // Earlier fragment should still have dominant value of 257.
+    CHECK(c_get_fragments_max_in_memory_tile_size(main_array_name_) == 257);
+
+    // Now want to consolidate, but not vacuum, max should still be 257.
+    tiledb::Array::consolidate(ctx_, main_array_name_);
+    // After consolidation, old fragment should still be there with 257.
+    CHECK(c_get_fragments_max_in_memory_tile_size(main_array_name_) == 257);
+
+    // Sparse/dense vacuum/consolidate semantics differ, vacuum dense eliminates
+    // old data.  After Vacuum, max should now be 42.
+    tiledb::Array::vacuum(ctx_, main_array_name_);
+    CHECK(c_get_fragments_max_in_memory_tile_size(main_array_name_) == 42);
+  }
+}
+
+TEST_CASE_METHOD(
+    MaxTileSizeFx,
+    "C++ API: Max fragment size, sparse, variable string dimension, variable string attribute",
+    "[capi][max-tile-size][sparse][evolve]") {
+  remove_temp_dir(main_array_name_);
+
+  create_sparse_array_d1_var_a1_string(true);
+
+  std::string basic_key_data{"abcdefghijklmnopqrstuvwxyz"};
+
+  write_sparse_array_d1_var_a1_string(basic_key_data, {0}, "");
+
+
+  CHECK(c_get_fragments_max_in_memory_tile_size(main_array_name_) == 26);
+
+  // Writing 26 empty items.
+  write_sparse_array_d1_var_a1_string(
+      basic_key_data,
+      {0,  1,  2,  3,  4,  5,  6,  7,  8,  9,  10, 11, 12,
+       13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25},
+      "");
+  // 208 / 16 == 8 (bytes per offset, offsets dominate over data size)
+  CHECK(c_get_fragments_max_in_memory_tile_size(main_array_name_) == 208);
+
+  // Writing 24 empty items, 2 occupied items.
+  write_sparse_array_d1_var_a1_string(
+      basic_key_data,
+      {0,  1,  2,  3,  4,  5,  6,  7,  8,  9,  10, 11, 12,
+       13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25},
+      "AB");
+
+  CHECK(c_get_fragments_max_in_memory_tile_size(main_array_name_) == 208);
+
+}
+
+TEST_CASE_METHOD(
+    MaxTileSizeFx,
+    "C++ API: Max fragment size, sparse, fix dimension, variable "
+    "string attribute",
+    "[capi][max-tile-size][dense][evolve]") {
+  remove_temp_dir(main_array_name_);
+
+  create_dense_array_d1_int_a1_string(
+      {1, 27}, 1, false, TILEDB_ROW_MAJOR, TILEDB_ROW_MAJOR);
+  CHECK(c_get_fragments_max_in_memory_tile_size(main_array_name_) == 0);
+
+  // Zero-len data written at index 1.
+  write_dense_array_d1_int_a1_string("", {0}, {1, 1});
+
+  // Size of single offset (sizeof(uint64_t)) dominates.
+  CHECK(c_get_fragments_max_in_memory_tile_size(main_array_name_) == 8);
+
+  // Attribute is variable non-nullable, so the single offset (uint64_t) of 
+  // 8 bytes will be max with strings <= 8 bytes writing len 1 data at idx 1.
+  write_dense_array_d1_int_a1_string(
```

Comment:
Can you run clang-format on this PR? This doesn't look right.

Reply:
I have run clang format various times, and it did not change things I expected it to, but will try again.